projects/addon-charts/components/pie-chart/pie-chart.component.ts

```diff
@@ -130,15 +130,20 @@ export class TuiPieChartComponent {
     }
 
     @tuiPure
-    private getSegments(value: readonly number[]): readonly [number, number][] {
-        const total = sum(...value);
+    private getSum(value: readonly number[]): number {
+        return sum(...value);
+    }
 
+    @tuiPure
+    private getSegments(value: readonly number[]): readonly [number, number][] {
         return value
             .map((currentItem, currentIndex, array) =>
                 array.reduce(
                     (sum, item, index) =>
-                        index < currentIndex ? (item / total) * 360 + sum : sum,
-                    (currentItem / total) * 360,
+                        index < currentIndex
+                            ? (item / this.getSum(value)) * 360 + sum
+                            : sum,
+                    (currentItem / this.getSum(value)) * 360,
```

Comment:
can we move `(currentItem / this.getSum(value)) * 360,` to another function? for reuse `(item / this.getSum(value)) * 360`

Reply:
Done